bot/exts/moderation/infraction/_utils.py

```diff
@@ -269,3 +272,44 @@ async def send_private_embed(user: MemberOrUser, embed: discord.Embed) -> bool:
             "The user either could not be retrieved or probably disabled their DMs."
         )
         return False
+
+
+class StaffBanConfirmationView(interactions.ViewWithUserAndRoleCheck):
+    """The confirmation view that is sent when a moderator attempts to ban a staff member."""
+
+    def __init__(
+        self,
+        *,
+        allowed_users: t.Sequence[int],
+        allowed_roles: t.Sequence[int],
+        timeout: t.Optional[float] = 180,
+        message: t.Optional[Message] = None
+    ) -> None:
+        super().__init__(
+            allowed_users=allowed_users,
+            allowed_roles=allowed_roles,
+            timeout=timeout,
+            message=message
+        )
+
+        self.value: t.Optional[bool] = None
+
+    async def send(self, channel: TextChannel, *, message_content: str) -> None:
+        """Send this view to the given channel with the given message content."""
+        self.message = await channel.send(message_content, view=self)
+
+    @discord.ui.button(label="Confirm", style=ButtonStyle.green)
+    async def confirm(self, interaction: Interaction, button: Button) -> None:
+        """Callback coroutine that is called when the "confirm" button is pressed."""
```

Comment:
Question: Can't the code used in `confirm` and `cancel` be moved to  a "private" method instead ? 
It's so similair, and the only thing changing is the value
```py
async __set_value(self, interaction: Interaction, button: Button, value: bool):
        await interaction.response.defer()
        self.value = value
        await super().on_timeout()
        self.stop()
   
@discord.ui.button(label="Cancel", style=ButtonStyle.red)
    async def cancel(self, interaction: Interaction, button: Button) -> None:
        """Callback coroutine that is called when the "cancel" button is pressed."""
        await self.__set_value(intraction, button, False)
 ```

Reply:
It definitely could! Though in this case I'm not 100% sure if it's necessary to do so. It would be relatively little effort to extract the common logic into it's own function, but only 2 functions are sharing this logic.

Reply:
I wouldn't call it necessary of course.

The decision is totally up to you, thanks!